akka-docs/rst/java/code/docs/http/javadsl/server/directives/FileUploadDirectivesExamplesTest.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (C) 2016-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package docs.http.javadsl.server.directives;
+
+import akka.http.impl.engine.rendering.BodyPartRenderer;
+import akka.http.javadsl.model.*;
+import akka.http.javadsl.server.Route;
+import akka.http.javadsl.server.Unmarshaller;
+import akka.http.javadsl.server.directives.FileInfo;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+import akka.stream.javadsl.Framing;
+import akka.stream.javadsl.Source;
+import akka.util.ByteString;
+import akka.util.ByteString$;
+import org.junit.Test;
+import java.io.File;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+import java.util.function.BiFunction;
+
+public class FileUploadDirectivesExamplesTest extends JUnitRouteTest {
+
+    @Test
+    public void testUploadedFile() {
+        //#uploadedFile
+        // function (FileInfo, File) => Route to process the file metadata and file itself
+        BiFunction<FileInfo, File, Route> infoFileRoute =
+                (info, file) -> {
+                    // do something with the file and file metadata ...
+                    file.delete();
+                    return complete(StatusCodes.OK);
+                };
+
+
+        final Route route = uploadedFile("csv", infoFileRoute);
+
+        Map<String, String> filenameMapping = new HashMap<>();
+        filenameMapping.put("filename", "data.csv");
+
+        akka.http.javadsl.model.Multipart.FormData multipartForm =
+                HttpEntities.fromParts(HttpEntities.createStrict("csv",
+                        HttpEntities.create(ContentTypes.TEXT_PLAIN_UTF8,
+                        "1,5,7\n11,13,17"), filenameMapping));
+
+        // test:
+        testRoute(route).run(HttpRequest.POST("/").withEntity(
+                multipartForm.toEntity(HttpCharsets.UTF_8, BodyPartRenderer
+                .randomBoundary(BodyPartRenderer.randomBoundary$default$1(),
+                        BodyPartRenderer.randomBoundary$default$2()))))
```

Comment:
ouch, missing api still, can we make it work without touching scala generated methods?